### ISSUE TITLE
Still the blank-only-'Connected' UI

### ISSUE THREAD

**Issue by ericBcreator** [s00500/ESPUI#209]
I thought I read this issue was tackled but getting back to my [clock project](https://www.hackster.io/ericBcreator/alarm-clock-with-web-interface-wake-up-light-temp-more-75dc6a) and recompiling with the latest updated libraries I still run into this issue (Wemos D1 mini):

![espui-err-22116](https://user-images.githubusercontent.com/44906396/200181805-0f4c2fe6-694c-4577-8318-ff05743f5b99.png)

**Comment by thomastech** [comment]
I looked at your Mar-2021 project code and can see you were using the SPIFFS file system. I assume you've revised your code to use the LITTLEFS method [ESPUI.beginLITTLEFS()] instead of SPIFFS. And have updated your build platform's file system folder with the *latest* web related files needed to run ESPUI's revised filesystem AND have uploaded them to the ESP32 device.

So assuming you've done all that and there's still no-joy, it's time for a simple experiment. I suggest you temporarily change to the built-in file server [ESPUI.begin()]. If this works then there is something wrong with the LITTLEFS filesystem. Could be missing web related files in your project, a bug in the Arduino IDE's file system uploader, or a problem with the current ESPUI library.

- Thomas

**Comment by ericBcreator** [comment]
Hi Tomas,

I updated to ESPUI 2.2.1 and the other libraries (it depends on) to their latest versions.
Recompiled (cleared all flash memory) and tried both with and without using the filesystem (LITTELEFS): no difference, still the same issue.

Experimenting in the past seemed to indicate the number of controls (>60) affect this issue.
So in the project I added an option to limit the number of controls.
But it is unreliable, sometimes the web ui will display, sometimes not.
And it's also browser unrelated: the same issue with Firefox, Chrome, Edge and Safari.

BTW: it is only with ESP8266 boards, I have no problems with ESP32s:

![ebc-32-221108](https://user-images.githubusercontent.com/44906396/200578882-d84a7613-a792-416d-b1a7-d7f558b37084.png)

**Comment by thomastech** [comment]
That is indeed an interesting problem. Maybe Firefox's F12->Console debugger will provide a clue on what is going on. This will show you the messages between the browser and ESPUI.

- Thomas

**Comment by ericBcreator** [comment]
F12 unfortunately provides no useful info. (Also) check this: the UI appearing with the Edge browser but a minute later blank with Firefox:

![afbeelding](https://user-images.githubusercontent.com/44906396/200649104-b7882b72-94e7-4f5c-8de8-8999bd9fb29a.png)

BTW: I mentioned this issue already 2 years ago, check this thread: https://github.com/s00500/ESPUI/issues/92
It seems Ian was working on a fix. Apparently it didn't work or it didn't made it into the latest version.

**Comment by ericBcreator** [comment]
Another BTW: one thing I noticed is sometimes the UI displays 'Control Connected' and other times '#SSID# Connected'.

**Comment by thomastech** [comment]
Your previous Issue #92 provided helpful background info. I'd have to agree that switching the ESP8266 to a ESP32 is the most practical solution when a lot of controls are used.

- Thomas

**Comment by ericBcreator** [comment]
Yes and the ESP32 Mini is almost a drop in replacement for my project.
I have built several with an ESP8266 though, so it would be nice if the UI
also worked without issues.

And an issue I had with ESP32s was that sometimes a brownout error
triggered when connecting to Wifi causing the EEPROM settings to be erased.
So both boards have their pros and cons.

**Comment by ericBcreator** [comment]
IIRC a while ago I read an e-mail mentioning that someone (Ian?) figured
out what was causing the blank screen and found another method of sending
the JSON data?
Not sure about it though.

**Comment by ericBcreator** [comment]
You're right about using correct power and decent cables.
In hindsight I have to say the brownout errors didn't occur since I added a
good power supply next to the PC USB connection (when working on the
project).

*The current ESPUI release sends data in much smaller chunks*
So I was right some changes were made (by MartinMueller2003).
In the past I did some retracing in the ESPAsync libraries but quit as it
took a lof of time with no results (yet) and it was becoming a big
sidetrack from my project.

What I don't get is why - with the exact same setup (board and code) -
sometimes the UI is blank and other times not (like in the picture I
posted).
That doesn't make sense, the logic program excution should be the same
every time.
It is not like it's running on a bloated OS like Windows with all kind of
other processes ready to mess things up.

**Comment by ericBcreator** [comment]
Fair points.

BTW: a good debugger would make life lot a lot easier,
Putting debug prints in the Arduino IDE is a pain in the but.
I have some ideas, will give them a try when I have some time.

**Comment by Eisbaeeer** [comment]
Got same issue with d1_mini_pro esp8266.
The Edge browser showing connected and the blank screen. The same with mobile phone.
Greets Lars

**Comment by d-a-v** [comment]
> But don't give up. It would be great if you or some other ESPUI user found the reason to this issue and published the fix. In the meantime I think swapping the ESP8266 with a ESP32 is a somewhat painless solution.

I think #241 would help.

**Comment by Looki2000** [comment]
Im having the same problem with ESPUI 2.2.4 and ArduinoJson 6.10.1 but on ESP32. Is there any fix yet?

**Comment by ericBcreator** [comment]
I got back this project around 1,5 month ago, recompiled it with current libraries and it seems it finally fixed the issues with the ESP8266 and ESPUI interface - for me. Still working since 6 weeks without an issue:

![scs24102301](https://github.com/user-attachments/assets/fee85c2a-bbc8-4400-a842-746b6c2e8c85)
